**Group Members:** Jojo Hu (47939335), guys add your names

In [11]:
library(tidyverse)

First, we load the data, taking care to convert the `experience` and `gender` columns to factors.

In [15]:
players <- read_csv("https://raw.githubusercontent.com/Yh194/ubc-dsci100-21/refs/heads/main/data/players.csv") |>
    mutate(across(c(experience, gender), as.factor))
# order experience levels
players$experience <- factor(players$experience, levels = c('Beginner', 'Amateur', 'Regular', 'Pro', 'Veteran'))

# list possible values for factors and their distribution
table(players$gender)
table(players$experience)

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.



          Agender            Female              Male        Non-binary 
                2                37               124                15 
            Other Prefer not to say      Two-Spirited 
                1                11                 6 


Beginner  Amateur  Regular      Pro  Veteran 
      35       63       36       14       48 

The `players.csv` file is a comma-separated CSV file with 7 columns and 196 rows.

### Specification

| Column Name      | Data Type | Meaning                                                   | Possible values/constraints                        |
|------------------|-----------|-----------------------------------------------------------|----------------------------------------------------|
| **experience**   | `fct`     | How much experience the player has in the game            | 'Beginner', 'Amateur', 'Regular', 'Pro', 'Veteran' |
| **subscribe**    | `lgl`     | Whether or not the player is subscribed to the newsletter |                                                    |
| **hashedEmail**  | `chr`     | The hash of the player's email, serving as a unique ID    |                                                    |
| **played_hours** | `dbl`     | The number of hours the player has played for             | Non-negative                                       |
| **name**         | `chr`     | The player's name                                         |                                                    |
| **gender**       | `fct`     | The player's gender                                       | 'Male', 'Female', 'Non-binary', 'Prefer not to say', 'Agender', 'Two-Spirited', 'Other' |
| **Age**          | `dbl`     | The player's age                                          | Non-negative                                       |

### Irrelevant Variables
`name` and `hashedEmail` are unique for each player, so they are not useful in our analysis.

### Potential Issues
- Played hours may be inaccurate because players might stay idle/AFK within the game, which would lead to an artificially inflated playtime
- Some players have disproportionately high playtime compared to others, which could skew averages
- Players could lie about or omit information such as age, gender, etc.
- There are a lot more male players (124) compared to other genders, which could introduce bias
- There are `NA` values present in the age column
- Skill level/experience is a very subjective column
  - Different people may have different opinions about what level of skill constitutes a "Pro" for example
  - The ranking is ambiguous, you can guess which value is more skilled than another but it's not clearly defined